unlock-app/src/components/content/events-collection/settings/General.tsx

```diff
@@ -0,0 +1,258 @@
+          </div>
+        </SettingCard>
+
+        {/* Banner settings */}
+        <SettingCard
+          label="Banner"
+          description="Update the banner of your event collection."
+        >
+          <Controller
+            name="banner"
+            control={methods.control}
+            rules={{ required: 'Banner is required' }}
+            render={({ field: { onChange, value }, fieldState: { error } }) => (
+              <ImageUpload
+                description="This banner will be used as the banner for your event collection. The recommended aspect ratio is 1400x300."
+                isUploading={isUploading}
+                size="full"
+                imageRatio="cover"
+                preview={value || eventCollection.banner!}
+                onChange={async (fileOrFileUrl: any) => {
+                  if (typeof fileOrFileUrl === 'string') {
+                    onChange(fileOrFileUrl)
+                  } else {
+                    const items = await uploadImage(fileOrFileUrl[0])
+                    const image = items?.[0]?.publicUrl
+                    if (!image) return
+                    onChange(image)
+                  }
+                }}
+                error={error?.message}
+              />
+            )}
+          />
+          <div className="flex flex-col sm:flex-row-reverse w-full pt-8">
+            <Button
+              loading={methods.formState.isSubmitting}
+              type="submit"
+              className="w-full sm:w-48"
+            >
+              Save
+            </Button>
+          </div>
+        </SettingCard>
+
+        {/* Links settings */}
+        <SettingCard
+          label="Links"
+          description="Update the links of your event collection."
+        >
+          {linkFields.map((field, index) => (
+            <EditLinkField key={field.id} index={index} remove={remove} />
+          ))}
+          <div className="flex justify-end">
+            <Button
+              type="button"
+              onClick={() => append({ type: 'website', url: '' })}
+              disabled={
+                linkFields.length > 0
+                  ? !links[linkFields.length - 1]?.type ||
+                    !links[linkFields.length - 1]?.url
+                  : false
+              }
+            >
+              Add Link
+            </Button>
+          </div>
+          {methods.formState.errors.links &&
+            typeof methods.formState.errors.links.message === 'string' && (
+              <p className="text-red-500 text-sm">
+                {methods.formState.errors.links.message}
+              </p>
+            )}
+          {/* Save button for Links */}
+          <div className="flex flex-end w-full pt-8 flex-row-reverse">
+            <Button
+              loading={methods.formState.isSubmitting}
+              type="submit"
+              className="w-48"
+            >
+              Save Links
+            </Button>
+          </div>
```

Comment:
This is not needed, let's auto-save when links are added or removed.

Reply:
done!